src/builtins.ts

```diff
@@ -771,6 +772,62 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      compiler.currentType = compiler.program.stringInstance.type;
+
+      if (checkTypeOptional(typeArguments, reportNode, compiler)
+        | checkArgsOptional(operands, 0, 1, reportNode, compiler)) {
+        compiler.error(
+          DiagnosticCode.Operation_not_supported,
+          reportNode.typeArgumentsRange
+        );
+        return module.unreachable();
+      }
+
+      let expression = new StringLiteralExpression();
+      let resultType: Type | null = null;
+
+      if (typeArguments !== null && typeArguments.length > 0) {
+        resultType = typeArguments[0];
+      } else {
+        resultType = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      }
```

Comment:
Most of the above can be simplified to just `evaluateConstantType`. See for example `isInteger`.

Reply:
Wow. That's really easy. Thanks for the tip. I spent a lot of time trying to find something like that.

Reply:
All set